src/Npgsql/NpgsqlDataReader.cs

```diff
@@ -1619,9 +1618,12 @@ T GetFieldValueCore<T>(int ordinal)
         if (typeof(T) == typeof(TextReader))
             PgReader.ThrowIfStreamActive();
 
-        var columnLength = SeekToColumn(async: false, ordinal, field).GetAwaiter().GetResult();
-        if (columnLength == -1)
-            return DbNullValueOrThrow<T>(field);
+        var columnLength =
+            _isSequential
```

Comment:
We do not use `SeekToColumn` here so we don't have to call `GetAwaiter().GetResult()` for `SeekToColumnNonSequential`?

Reply:
Yes it was to save on having to wrap SeekToColumnNonSequential and immediately undo it again

Reply:
OK. Not that amazing but OK.

Reply:
I know async in general will get some more love soon from the runtime and C# language team. 
Probably won't help much with this level of optimization but hopefully the impact of introducing async methods etc will be reduced somewhat. They're also going to consider binary size improvements which might help reduce the need for our horrible workaround code.